File: spoj/classicals/ACPC10D.cpp
#include <iostream>

using namespace std;

long long **a;

long long min(long long a, long long b){
	if (a <= b) return a;
	return b;
}

long long min(long long a, long long b, long long c){
	if (a <= b && a <= c) return a;
	if (b <= c ) return b;
	return c;
}

long long solve(long long n, int i, int j){

}

int main(void){
	long long n;
	cin >> n;
	long long test = 1;
	do{

		a = new long long*[n];
		for (long long i = 0; i < n; i += 1){
			a[i] = new long long[3];
		}
		for (long long i = 0; i < n; i += 1){
			for (long long j = 0; j < 3; j += 1){
				cin >> a[i][j];
			}
		}
		cout << test << ". " << solve(a, n) << endl;

		cin >> n;
		test += 1;
		delete[] a;
	}while(n);

}

File: spoj/classicals/ACPC10D.py

def solve(a, n):
	for i in xrange(3):
		a[1][i] += a[0][1];
	for i in xrange(2,n):
		for j in xrange(3):
			if j == 0:
				a[i][j] += min(a[i-1][j],a[i-1][j+1])
			elif j == 2:
				a[i][j] += min(a[i-1][j],a[i-1][j-1])
			else:
				a[i][j] += min(a[i-1][j],a[i-1][j-1],a[i-1][j+1])
	# for s in a:
	# 	print s
	return a[n-1][1]
test = 1
import sys
inputs = sys.stdin.read()
inputs = inputs.split("\n")
line = 0
while 1:
	n = int(inputs[line])
	line += 1
	if n == 0: break
	a = []
	for i in xrange(n):
		a.append(map(int, inputs[line].split()))
		line += 1
	print "%d. %d"%(test,solve(a,n))
	test += 1